segfault when loading D64, G64 or NIB image

Using current retroarch 1.9.0 with core "VICE x64 (3.3 e1e4bdd)" (from online updater) and bare configuration,
launching the core with an image ends with a segfault. I tested with a G64, a NIB and a D64 image.
Core without image is OK and with a cartridge too. I don't think my images are corrupted or so, they work well with 
anterior retroarch version (1.8.5 with core VICE x64 3.3 79292ad for example).

Logs:
[retroarch.log](https://github.com/libretro/vice-libretro/files/5726723/retroarch.log)

> [INFO] RetroArch 1.9.0 (Git 5e551dd92b)
> [INFO] === Build =======================================
> [INFO] CPU Model Name: Intel(R) Core(TM) i5 CPU         750  @ 2.67GHz
> [INFO] Capabilities:  MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE4.2
> [INFO] Built: Aug  8 2020
> [INFO] Version: 1.9.0
> [INFO] Git: 5e551dd92b
> [INFO] =================================================
> [INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
> [INFO] [Overrides]: Redirecting save file to "/home/ben/.config/retroarch/saves/.srm".
> [INFO] [Overrides]: Redirecting save state to "/home/ben/.config/retroarch/states/.state".
> [INFO] Version of libretro API: 1
> [INFO] Compiled against API: 1
> [INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
> [INFO] [Video]: Video @ 960x720
> [ERROR] [Wayland]: Failed to connect to Wayland server.
> [INFO] [GLX]: GLX_EXT_swap_control_tear supported.
> [INFO] [GL]: Found GL context: x
> [INFO] [GL]: Detecting screen resolution 2560x1440.
> [INFO] [GLX]: Window manager is GNOME Shell.
> [INFO] [GLX]: X = 0, Y = 0, W = 960, H = 720.
> [INFO] [GLX]: Found swap function: glXSwapIntervalEXT.
> [INFO] [GLX]: glXSwapInterval(1)
> [INFO] [GL]: Vendor: AMD, Renderer: Radeon RX550/550 Series (POLARIS12, DRM 3.39.0, 5.9.14-arch1-1, LLVM 11.0.0).
> [INFO] [GL]: Version: 4.6 (Compatibility Profile) Mesa 20.3.0.
> [INFO] [GL]: Using resolution 960x720
> [INFO] [GL]: Default shader backend found: glsl.
> [INFO] [Shader driver]: Using GLSL shader backend.
> [INFO] [GLSL]: Checking GLSL shader support ...
> [WARN] [GL]: Stock GLSL shaders will be used.
> [INFO] [GLSL]: Found GLSL vertex shader.
> [INFO] [GLSL]: Found GLSL fragment shader.
> [INFO] [GLSL]: Linking GLSL program.
> [INFO] [GLSL]: Found GLSL vertex shader.
> [INFO] [GLSL]: Found GLSL fragment shader.
> [INFO] [GLSL]: Linking GLSL program.
> [INFO] [GLSL]: Found GLSL vertex shader.
> [INFO] [GLSL]: Found GLSL fragment shader.
> [INFO] [GLSL]: Linking GLSL program.
> [INFO] [GL]: Using 4 textures.
> [INFO] [GL]: Loaded 1 program(s).
> [INFO] [GL]: Using GL_RGB565 for texture uploads.
> [INFO] [udev]: Pad #0 (/dev/input/event7) supports force feedback.
> [INFO] [udev]: Pad #0 (/dev/input/event7) supports 16 force feedback effects.
> [INFO] [udev]: Pad #1 (/dev/input/event8) supports force feedback.
> [INFO] [udev]: Pad #1 (/dev/input/event8) supports 16 force feedback effects.
> [INFO] [Joypad]: Found joypad driver: "udev".
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [DBus]: Suspended screensaver via DBus.
> [INFO] [Video]: Found display server: x11
> [INFO] [PulseAudio]: Requested 24576 bytes buffer, got 35696.
> [INFO] [Display]: Found display driver: "gl".
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Display]: Found display driver: "gl".
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [MIDI]: Input disabled.
> [INFO] [MIDI]: Output disabled.
> [INFO] [MIDI]: Initialized "alsa" driver.
> [INFO] Disconnecting device from port 6.
> [INFO] Disconnecting device from port 7.
> [INFO] Disconnecting device from port 8.
> [INFO] Disconnecting device from port 9.
> [INFO] Disconnecting device from port 10.
> [INFO] Disconnecting device from port 11.
> [INFO] Disconnecting device from port 12.
> [INFO] Disconnecting device from port 13.
> [INFO] Disconnecting device from port 14.
> [INFO] Disconnecting device from port 15.
> [INFO] Disconnecting device from port 16.
> [INFO] [SRAM]: SRAM will not be saved.
> [INFO] [Playlist]: Loading history file: [/home/ben/.config/retroarch/content_history.lpl].
> [INFO] [Playlist]: Loading history file: [/home/ben/.config/retroarch/content_music_history.lpl].
> [INFO] [Playlist]: Loading history file: [/home/ben/.config/retroarch/content_video_history.lpl].
> [INFO] [Playlist]: Loading history file: [/home/ben/.config/retroarch/content_image_history.lpl].
> [INFO] [Playlist]: Loading favorites file: [/home/ben/.config/retroarch/content_favorites.lpl].
> [INFO] [GL]: VSync => on
> [INFO] [GLX]: glXSwapInterval(1)
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [GL]: VSync => on
> [INFO] [GLX]: glXSwapInterval(1)
> [INFO] [PulseAudio]: Pausing.
> [INFO] [CONTENT LOAD]: Updating firmware status for: /home/ben/.config/retroarch/cores/vice_x64_libretro.so on /home/ben/.config/retroarch/system
> [INFO] [CORE]: Using content: /home/ben/pkg/c64/no-intro/arka.g64.
> [INFO] Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
> [INFO] [CORE]: Unloading core..
> [INFO] [CORE]: Unloading core symbols..
> [INFO] [XINERAMA]: Xinerama version: 1.1.
> [INFO] [XINERAMA]: Xinerama screens: 1.
> [INFO] [XINERAMA]: Saved monitor #0.
> [INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
> [INFO] RetroArch 1.9.0 (Git 5e551dd92b)
> [INFO] [Overrides]: Redirecting save file to "/home/ben/.config/retroarch/saves/arka.srm".
> [INFO] [Overrides]: Redirecting save state to "/home/ben/.config/retroarch/states/arka.state".
> [INFO] === Build =======================================
> [INFO] CPU Model Name: Intel(R) Core(TM) i5 CPU         750  @ 2.67GHz
> [INFO] Capabilities:  MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE4.2
> [INFO] Built: Aug  8 2020
> [INFO] Version: 1.9.0
> [INFO] Git: 5e551dd92b
> [INFO] =================================================
> [INFO] [CORE]: Loading dynamic libretro core from: "/home/ben/.config/retroarch/cores/vice_x64_libretro.so"
> [INFO] [Overrides]: No core-specific overrides found at /home/ben/.config/retroarch/config/VICE x64/VICE x64.cfg.
> [INFO] [Overrides]: No content-dir-specific overrides found at /home/ben/.config/retroarch/config/VICE x64/no-intro.cfg.
> [INFO] [Overrides]: No game-specific overrides found at /home/ben/.config/retroarch/config/VICE x64/arka.cfg.
> [INFO] [Environ]: SET_CONTROLLER_INFO.
> [INFO] [Environ]: GET_CORE_OPTIONS_VERSION.
> [INFO] [Environ]: SET_CORE_OPTIONS.
> [INFO] [Environ]: SET_SUPPORT_NO_GAME: yes.
> [INFO] [Remaps]: remap directory: /home/ben/.config/retroarch/config/remaps
> [INFO] [Overrides]: Redirecting save file to "/home/ben/.config/retroarch/saves/arka.srm".
> [INFO] [Overrides]: Redirecting save state to "/home/ben/.config/retroarch/states/arka.state".
> [INFO] [Environ]: GET_LOG_INTERFACE.
> [INFO] [Environ]: SYSTEM_DIRECTORY: "/home/ben/.config/retroarch/system".
> [INFO] [Environ]: CORE_ASSETS_DIRECTORY: "/home/ben/.config/retroarch/downloads".
> [INFO] [Environ]: GET_SAVE_DIRECTORY.
> [INFO] [Environ]: GET_DISK_CONTROL_INTERFACE_VERSION.
> [INFO] [Environ]: SET_DISK_CONTROL_EXT_INTERFACE.
> [INFO] [Environ]: SET_SERIALIZATION_QUIRKS.
> [INFO] [Environ]: SET_INPUT_DESCRIPTORS:
> [INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
> [INFO] [Environ]: SET_SUPPORT_ACHIEVEMENTS: yes.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [CONTENT LOAD]: Content loading skipped. Implementation will load it on its own.
> [libretro INFO] Arg0: x64
> [libretro INFO] Arg1: /home/ben/pkg/c64/no-intro/arka.g64
> [libretro INFO] --------------------------------------------------------------------------------
> [libretro INFO] VICE 3.3
> [libretro INFO] Git commit: e1e4bdd
> [libretro INFO] Build date: Nov 16 2020 18:55:50
> [libretro INFO] 
> [libretro INFO] Welcome to x64, the free portable C64 Emulator.
> [libretro INFO] 
> [libretro INFO] This is free software with ABSOLUTELY NO WARRANTY.
> [libretro INFO] --------------------------------------------------------------------------------
> [libretro ERROR] 2000 ROM image not found. Hardware-level 2000 emulation is not available.
> [libretro ERROR] 4000 ROM image not found. Hardware-level 4000 emulation is not available.
> [libretro INFO] Finished loading ROM images.
> [libretro INFO] Available sound devices: retro
> [libretro INFO] Resource Mouse = 1
> [libretro INFO] Resource Printer4 = 1
> [libretro INFO] Resource AutostartPrgMode = 1
> [libretro INFO] Resource AutostartDelayRandom = 0
> [libretro INFO] Resource SoundVolume = 0
> [libretro INFO] Resource VICIIPaletteFile = "colodore"
> [libretro INFO] Resource VICIIExternalPalette = 1
> [libretro INFO] Resource VICIIColorGamma = 2800
> [libretro INFO] Resource VICIIColorTint = 1000
> [libretro INFO] Resource VICIIColorSaturation = 1000
> [libretro INFO] Resource VICIIColorContrast = 1000
> [libretro INFO] Resource VICIIColorBrightness = 1000
> [libretro INFO] Resource UserportJoy = 0
> [libretro INFO] Resource AutostartWarp = 0
> [libretro INFO] Resource DriveTrueEmulation = 1
> [libretro INFO] Resource VirtualDevices = 1
> [libretro INFO] Resource AttachDevice8Readonly = 0
> [libretro INFO] Resource EasyFlashWriteCRT = 1
> [libretro INFO] Resource KernalName = "kernal"
> [libretro INFO] Resource DosName1541 = "dos1541"
> [libretro INFO] Resource DosName1571 = "dos1571"
> [libretro INFO] Resource DosName1581 = "dos1581"
> [libretro INFO] Resource DriveSoundEmulationVolume = 400
> [libretro INFO] Resource DriveSoundEmulation = 1
> [libretro INFO] Resource VICIIAudioLeak = 0
> [libretro INFO] Resource SFXSoundExpander = 0
> [libretro INFO] Resource SidResidSampling = 2
> [libretro INFO] Resource SidResidPassband = 90
> [libretro INFO] Resource SidResidGain = 97
> [libretro INFO] Resource SidResidFilterBias = 500
> [libretro INFO] Resource SidResid8580Passband = 90
> [libretro INFO] Resource SidResid8580Gain = 97
> [libretro INFO] Resource SidResid8580FilterBias = 1500
> [libretro INFO] Resource SidStereo = 0
> [libretro INFO] Resource REU = 0
> [libretro INFO] Autodetecting image type of `/home/ben/pkg/c64/no-intro/arka.g64'.
> [libretro INFO] Attached file `/home/ben/pkg/c64/no-intro/arka.g64' as a disk image.
> [libretro INFO] Autodetected image type 100.
> [libretro INFO] Resetting the machine to autostart '*'
> [libretro INFO] RESET.
> [libretro INFO] Opened device `retro', speed 48000Hz, fragment size 0.7ms, buffer size 2ms
> [libretro INFO] reSID: MOS6581, filter on, sampling rate 48000Hz - resampling, pass to 21600Hz
> [libretro INFO] `/home/ben/pkg/c64/no-intro/arka.g64' recognized as disk image.
> [libretro INFO] Main CPU: starting at ($FFFC).
> [libretro INFO] Main CPU: RESET.
> [INFO] Version of libretro API: 1
> [INFO] Compiled against API: 1
> [INFO] [Cheats]: Load game-specific cheatfile: /home/ben/.config/retroarch/cheats/VICE x64/arka.cht
> [INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
> [INFO] [Video]: Timings deviate too much. Will not adjust. (Display = 60.00 Hz, Game = 50.12 Hz)
> [INFO] [Video]: Video @ 864x816
> [ERROR] [Wayland]: Failed to connect to Wayland server.
> [INFO] [GLX]: GLX_EXT_swap_control_tear supported.
> [INFO] [GL]: Found GL context: x
> [INFO] [GL]: Detecting screen resolution 2560x1440.
> [INFO] [GLX]: Window manager is GNOME Shell.
> [INFO] [GLX]: X = 0, Y = 0, W = 864, H = 816.
> [INFO] [GLX]: Found swap function: glXSwapIntervalEXT.
> [INFO] [GLX]: glXSwapInterval(1)
> [INFO] [GL]: Vendor: AMD, Renderer: Radeon RX550/550 Series (POLARIS12, DRM 3.39.0, 5.9.14-arch1-1, LLVM 11.0.0).
> [INFO] [GL]: Version: 4.6 (Compatibility Profile) Mesa 20.3.0.
> [INFO] [GL]: Using resolution 864x816
> [INFO] [GL]: Default shader backend found: glsl.
> [INFO] [Shader driver]: Using GLSL shader backend.
> [INFO] [GLSL]: Checking GLSL shader support ...
> [WARN] [GL]: Stock GLSL shaders will be used.
> [INFO] [GLSL]: Found GLSL vertex shader.
> [INFO] [GLSL]: Found GLSL fragment shader.
> [INFO] [GLSL]: Linking GLSL program.
> [INFO] [GLSL]: Found GLSL vertex shader.
> [INFO] [GLSL]: Found GLSL fragment shader.
> [INFO] [GLSL]: Linking GLSL program.
> [INFO] [GLSL]: Found GLSL vertex shader.
> [INFO] [GLSL]: Found GLSL fragment shader.
> [INFO] [GLSL]: Linking GLSL program.
> [INFO] [GL]: Using 4 textures.
> [INFO] [GL]: Loaded 1 program(s).
> [INFO] [GL]: Using GL_RGB565 for texture uploads.
> [INFO] [udev]: Pad #0 (/dev/input/event7) supports force feedback.
> [INFO] [udev]: Pad #0 (/dev/input/event7) supports 16 force feedback effects.
> [INFO] [udev]: Pad #1 (/dev/input/event8) supports force feedback.
> [INFO] [udev]: Pad #1 (/dev/input/event8) supports 16 force feedback effects.
> [INFO] [Joypad]: Found joypad driver: "udev".
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [DBus]: Suspended screensaver via DBus.
> [INFO] [Video]: Found display server: x11
> [INFO] [PulseAudio]: Requested 24576 bytes buffer, got 35696.
> [INFO] [Display]: Found display driver: "gl".
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [libretro INFO] retro_set_environment: GET_CORE_OPTIONS_VERSION failed, not setting CORE_OPTIONS now.
> [libretro INFO] retro_set_environment: GET_CORE_OPTIONS_VERSION failed, not setting CORE_OPTIONS now.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [MIDI]: Input disabled.
> [INFO] [MIDI]: Output disabled.
> [INFO] [MIDI]: Initialized "alsa" driver.
> [INFO] Disconnecting device from port 6.
> [INFO] Disconnecting device from port 7.
> [INFO] Disconnecting device from port 8.
> [INFO] Disconnecting device from port 9.
> [INFO] Disconnecting device from port 10.
> [INFO] Disconnecting device from port 11.
> [INFO] Disconnecting device from port 12.
> [INFO] Disconnecting device from port 13.
> [INFO] Disconnecting device from port 14.
> [INFO] Disconnecting device from port 15.
> [INFO] Disconnecting device from port 16.
> [INFO] [SRAM]: SRAM will not be saved.
> [INFO] [Playlist]: Loading history file: [/home/ben/.config/retroarch/content_history.lpl].
> [INFO] [Playlist]: Loading history file: [/home/ben/.config/retroarch/content_music_history.lpl].
> [INFO] [Playlist]: Loading history file: [/home/ben/.config/retroarch/content_video_history.lpl].
> [INFO] [Playlist]: Loading history file: [/home/ben/.config/retroarch/content_image_history.lpl].
> [INFO] [Playlist]: Loading favorites file: [/home/ben/.config/retroarch/content_favorites.lpl].
> [INFO] [GL]: VSync => on
> [INFO] [GLX]: glXSwapInterval(1)
> [INFO] [Playlist]: Written to playlist file: /home/ben/.config/retroarch/content_history.lpl
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] [Font]: Using font rendering backend: freetype.
> [INFO] SET_GEOMETRY: 384x272, aspect: 1.322.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
> [libretro INFO] Resource Mouse = 1
> [libretro INFO] Resource Printer4 = 1
> [libretro INFO] Resource AutostartPrgMode = 1
> [libretro INFO] Resource AutostartDelayRandom = 0
> [libretro INFO] Resource SoundVolume = 0
> [libretro INFO] Resource VICIIPaletteFile = "colodore"
> [libretro INFO] Resource VICIIExternalPalette = 1
> [libretro INFO] Resource VICIIColorGamma = 2800
> [libretro INFO] Resource VICIIColorTint = 1000
> [libretro INFO] Resource VICIIColorSaturation = 1000
> [libretro INFO] Resource VICIIColorContrast = 1000
> [libretro INFO] Resource VICIIColorBrightness = 1000
> [libretro INFO] Resource UserportJoy = 0
> [libretro INFO] Resource AutostartWarp = 0
> [libretro INFO] Resource DriveTrueEmulation = 1
> [libretro INFO] Resource VirtualDevices = 1
> [libretro INFO] Resource AttachDevice8Readonly = 0
> [libretro INFO] Resource EasyFlashWriteCRT = 1
> [libretro INFO] Resource KernalName = "kernal"
> [libretro INFO] Resource DosName1541 = "dos1541"
> [libretro INFO] Resource DosName1571 = "dos1571"
> [libretro INFO] Resource DosName1581 = "dos1581"
> [libretro INFO] Resource DriveSoundEmulationVolume = 400
> [libretro INFO] Resource DriveSoundEmulation = 1
> [libretro INFO] Resource VICIIAudioLeak = 0
> [libretro INFO] Resource SFXSoundExpander = 0
> [libretro INFO] Resource SidResidSampling = 2
> [libretro INFO] Resource SidResidPassband = 90
> [libretro INFO] Resource SidResidGain = 97
> [libretro INFO] Resource SidResidFilterBias = 500
> [libretro INFO] Resource SidResid8580Passband = 90
> [libretro INFO] Resource SidResid8580Gain = 97
> [libretro INFO] Resource SidResid8580FilterBias = 1500
> [libretro INFO] Resource SidStereo = 0
> [libretro INFO] Resource REU = 0
> [libretro INFO] Arg0: x64
> [libretro INFO] Arg1: /home/ben/pkg/c64/no-intro/arka.g64
> [libretro INFO] Autodetecting image type of `/home/ben/pkg/c64/no-intro/arka.g64'.
> [libretro INFO] Attached file `/home/ben/pkg/c64/no-intro/arka.g64' as a disk image.
> [libretro INFO] Autodetected image type 100.
> [libretro INFO] Resetting the machine to autostart '*'
> [libretro INFO] `/home/ben/pkg/c64/no-intro/arka.g64' recognized as disk image.
> [libretro INFO] Image for autostart: '/home/ben/pkg/c64/no-intro/arka.g64'
> Segmentation fault (core dumped)

--- a/libretro/libretro-core.c
+++ b/libretro/libretro-core.c
@@ -3277,27 +3277,30 @@ void retro_set_environment(retro_environment_t cb)
                retro_system_data_directory, FSDEV_DIR_SEP_STR, machine_name);
 
          /* Scan system/vice/machine directory for cartridges */
-         cart_dir = opendir(machine_directory);
-         while ((cart_dirp = readdir(cart_dir)) != NULL && j < RETRO_NUM_CORE_OPTION_VALUES_MAX - 1)
+         if (path_is_directory(machine_directory))
          {
-             /* Blacklisted */
-             if (!strcmp(cart_dirp->d_name, "scpu-dos-1.4.bin") ||
-                 !strcmp(cart_dirp->d_name, "scpu-dos-2.04.bin"))
-                 continue;
+            cart_dir = opendir(machine_directory);
+            while ((cart_dirp = readdir(cart_dir)) != NULL && j < RETRO_NUM_CORE_OPTION_VALUES_MAX - 1)
+            {
+                /* Blacklisted */
+                if (!strcmp(cart_dirp->d_name, "scpu-dos-1.4.bin") ||
+                    !strcmp(cart_dirp->d_name, "scpu-dos-2.04.bin"))
+                    continue;
 
-             if (dc_get_image_type(cart_dirp->d_name) == DC_IMAGE_TYPE_MEM)
-             {
-                 char cart_path[RETRO_PATH_MAX] = {0};
-                 char cart_label[50] = {0};
-                 snprintf(cart_path, sizeof(cart_path), "%s", cart_dirp->d_name);
-                 snprintf(cart_label, sizeof(cart_label), "%s", path_remove_extension(cart_dirp->d_name));
+                if (dc_get_image_type(cart_dirp->d_name) == DC_IMAGE_TYPE_MEM)
+                {
+                    char cart_value[RETRO_PATH_MAX] = {0};
+                    char cart_label[50] = {0};
+                    snprintf(cart_value, sizeof(cart_value), "%s", cart_dirp->d_name);
+                    snprintf(cart_label, sizeof(cart_label), "%s", path_remove_extension(cart_dirp->d_name));
 
-                 core_options[i].values[j].value = strdup(cart_path);
-                 core_options[i].values[j].label = strdup(cart_label);
-                 ++j;
-             }
+                    core_options[i].values[j].value = strdup(cart_value);
+                    core_options[i].values[j].label = strdup(cart_label);
+                    ++j;
+                }
+            }
+            closedir(cart_dir);
          }
-         closedir(cart_dir);
 
          core_options[i].values[j].value = NULL;
          core_options[i].values[j].label = NULL;

--- a/libretro/libretro-core.c
+++ b/libretro/libretro-core.c
@@ -3411,7 +3411,7 @@ static void update_variables(void)
       char cart_full[RETRO_PATH_MAX] = {0};
 
       if (!strcmp(var.value, "none"))
-         snprintf(cart_full, sizeof(cart_full), "");
+         snprintf(cart_full, sizeof(cart_full), "%s", "");
       else
          snprintf(cart_full, sizeof(cart_full), "%s%s%s%s%s",
                retro_system_data_directory, FSDEV_DIR_SEP_STR, machine_name, FSDEV_DIR_SEP_STR, var.value);

--- a/libretro/libretro-core.c
+++ b/libretro/libretro-core.c
@@ -586,7 +586,7 @@ static int process_cmdline(const char* argv)
         /* ZIP + NIB vars, use the same temp directory for single NIBs */
         char zip_basename[RETRO_PATH_MAX] = {0};
         snprintf(zip_basename, sizeof(zip_basename), "%s", path_basename(full_path));
-        snprintf(zip_basename, sizeof(zip_basename), "%s", path_remove_extension(zip_basename));
+        path_remove_extension(zip_basename);
 
         char nib_input[RETRO_PATH_MAX] = {0};
         char nib_output[RETRO_PATH_MAX] = {0};
@@ -747,7 +747,7 @@ static int process_cmdline(const char* argv)
                     case CARTRIDGE_VIC20_MEGACART:
                         snprintf(cartmega_temp, sizeof(cartmega_temp), "%s", argv);
                         snprintf(cartmega_temp, sizeof(cartmega_temp), "%s", path_basename(cartmega_temp));
-                        snprintf(cartmega_temp, sizeof(cartmega_temp), "%s", path_remove_extension(cartmega_temp));
+                        path_remove_extension(cartmega_temp);
                         snprintf(cartmega_nvram, sizeof(cartmega_nvram), "%s%s%s%s",
                                  retro_save_directory, FSDEV_DIR_SEP_STR, cartmega_temp, ".nvr");
                         Add_Option("-mcnvramfile");
@@ -756,21 +756,21 @@ static int process_cmdline(const char* argv)
                         break;
                     case -1: /* Separate ROM combination shenanigans, Gamebase style */
                         snprintf(cart_2000, sizeof(cart_2000), "%s", argv);
-                        snprintf(cart_2000, sizeof(cart_2000), "%s", path_remove_extension(cart_2000));
+                        path_remove_extension(cart_2000);
                         snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring((const char*)cart_2000, "-2000", ""));
                         snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring((const char*)cart_2000, "-6000", ""));
                         snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring((const char*)cart_2000, "-a000", ""));
                         snprintf(cart_2000, sizeof(cart_2000), "%s%s%s", cart_2000, "-2000", ".prg");
 
                         snprintf(cart_6000, sizeof(cart_6000), "%s", argv);
-                        snprintf(cart_6000, sizeof(cart_6000), "%s", path_remove_extension(cart_6000));
+                        path_remove_extension(cart_6000);
                         snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring((const char*)cart_6000, "-2000", ""));
                         snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring((const char*)cart_6000, "-6000", ""));
                         snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring((const char*)cart_6000, "-a000", ""));
                         snprintf(cart_6000, sizeof(cart_6000), "%s%s%s", cart_6000, "-6000", ".prg");
 
                         snprintf(cart_A000, sizeof(cart_A000), "%s", argv);
-                        snprintf(cart_A000, sizeof(cart_A000), "%s", path_remove_extension(cart_A000));
+                        path_remove_extension(cart_A000);
                         snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring((const char*)cart_A000, "-2000", ""));
                         snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring((const char*)cart_A000, "-6000", ""));
                         snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring((const char*)cart_A000, "-a000", ""));
@@ -813,28 +813,28 @@ static int process_cmdline(const char* argv)
                         }
 
                         snprintf(cart_2000, sizeof(cart_2000), "%s", argv);
-                        snprintf(cart_2000, sizeof(cart_2000), "%s", path_remove_extension(cart_2000));
+                        path_remove_extension(cart_2000);
                         snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring((const char*)cart_2000, "[4000]", "[2000]"));
                         snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring((const char*)cart_2000, "[6000]", "[2000]"));
                         snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring((const char*)cart_2000, "[A000]", "[2000]"));
                         snprintf(cart_2000, sizeof(cart_2000), "%s%s", cart_2000, ".crt");
 
                         snprintf(cart_4000, sizeof(cart_4000), "%s", argv);
-                        snprintf(cart_4000, sizeof(cart_4000), "%s", path_remove_extension(cart_4000));
+                        path_remove_extension(cart_4000);
                         snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring((const char*)cart_4000, "[2000]", "[4000]"));
                         snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring((const char*)cart_4000, "[6000]", "[4000]"));
                         snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring((const char*)cart_4000, "[A000]", "[4000]"));
                         snprintf(cart_4000, sizeof(cart_4000), "%s%s", cart_4000, ".crt");
 
                         snprintf(cart_6000, sizeof(cart_6000), "%s", argv);
-                        snprintf(cart_6000, sizeof(cart_6000), "%s", path_remove_extension(cart_6000));
+                        path_remove_extension(cart_6000);
                         snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring((const char*)cart_6000, "[2000]", "[6000]"));
                         snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring((const char*)cart_6000, "[4000]", "[6000]"));
                         snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring((const char*)cart_6000, "[A000]", "[6000]"));
                         snprintf(cart_6000, sizeof(cart_6000), "%s%s", cart_6000, ".crt");
 
                         snprintf(cart_A000, sizeof(cart_A000), "%s", argv);
-                        snprintf(cart_A000, sizeof(cart_A000), "%s", path_remove_extension(cart_A000));
+                        path_remove_extension(cart_A000);
                         snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring((const char*)cart_A000, "[2000]", "[A000]"));
                         snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring((const char*)cart_A000, "[4000]", "[A000]"));
                         snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring((const char*)cart_A000, "[6000]", "[A000]"));
@@ -882,19 +882,19 @@ static int process_cmdline(const char* argv)
                         }
 
                         snprintf(cart_2000, sizeof(cart_2000), "%s", argv);
-                        snprintf(cart_2000, sizeof(cart_2000), "%s", path_remove_extension(cart_2000));
+                        path_remove_extension(cart_2000);
                         snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring((const char*)cart_2000, "$6000", "$2000"));
                         snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring((const char*)cart_2000, "$A000", "$2000"));
                         snprintf(cart_2000, sizeof(cart_2000), "%s%s", cart_2000, ".20");
 
                         snprintf(cart_6000, sizeof(cart_6000), "%s", argv);
-                        snprintf(cart_6000, sizeof(cart_6000), "%s", path_remove_extension(cart_6000));
+                        path_remove_extension(cart_6000);
                         snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring((const char*)cart_6000, "$2000", "$6000"));
                         snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring((const char*)cart_6000, "$A000", "$6000"));
                         snprintf(cart_6000, sizeof(cart_6000), "%s%s", cart_6000, ".60");
 
                         snprintf(cart_A000, sizeof(cart_A000), "%s", argv);
-                        snprintf(cart_A000, sizeof(cart_A000), "%s", path_remove_extension(cart_A000));
+                        path_remove_extension(cart_A000);
                         snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring((const char*)cart_A000, "$2000", "$A000"));
                         snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring((const char*)cart_A000, "$6000", "$A000"));
                         snprintf(cart_A000, sizeof(cart_A000), "%s%s", cart_A000, ".a0");

--- a/libretro/libretro-dc.c
+++ b/libretro/libretro-dc.c
@@ -479,7 +479,7 @@ bool dc_replace_file(dc_storage* dc, int index, const char* filename)
       /* ZIP + NIB vars, use the same temp directory for single NIBs */
       char zip_basename[RETRO_PATH_MAX] = {0};
       snprintf(zip_basename, sizeof(zip_basename), "%s", path_basename(full_path_replace));
-      snprintf(zip_basename, sizeof(zip_basename), "%s", path_remove_extension(zip_basename));
+      path_remove_extension(zip_basename);
 
       char nib_input[RETRO_PATH_MAX] = {0};
       char nib_output[RETRO_PATH_MAX] = {0};
@@ -923,7 +923,7 @@ void dc_parse_list(dc_storage* dc, const char* list_file, bool is_vfl, const cha
             /* ZIP + NIB vars, use the same temp directory for single NIBs */
             char zip_basename[RETRO_PATH_MAX] = {0};
             snprintf(zip_basename, sizeof(zip_basename), "%s", path_basename(full_path));
-            snprintf(zip_basename, sizeof(zip_basename), "%s", path_remove_extension(zip_basename));
+            path_remove_extension(zip_basename);
 
             char nib_input[RETRO_PATH_MAX] = {0};
             char nib_output[RETRO_PATH_MAX] = {0};

--- a/libretro/libretro-mapper.c
+++ b/libretro/libretro-mapper.c
@@ -152,7 +152,7 @@ void emu_function(int function)
          /* Lock current port */
          cur_port_locked = true;
          /* Statusbar notification */
-         snprintf(statusbar_text, 56, "%c Port %-50d",
+         snprintf(statusbar_text, 56, "%c Port %-48d",
                (' ' | 0x80), cur_port);
          imagename_timer = 50;
          break;
@@ -169,7 +169,7 @@ void emu_function(int function)
          /* Lock aspect ratio */
          opt_aspect_ratio_locked = true;
          /* Statusbar notification */
-         snprintf(statusbar_text, 56, "%c Pixel Aspect %-50s",
+         snprintf(statusbar_text, 56, "%c Pixel Aspect %-40s",
                (' ' | 0x80), (opt_aspect_ratio == 1) ? "PAL" : (opt_aspect_ratio == 2) ? "NTSC" : "1:1");
          imagename_timer = 50;
          break;
@@ -186,7 +186,7 @@ void emu_function(int function)
          /* Lock turbo fire */
          turbo_fire_locked = true;
          /* Statusbar notification */
-         snprintf(statusbar_text, 56, "%c Turbo Fire %-50s",
+         snprintf(statusbar_text, 56, "%c Turbo Fire %-42s",
                (' ' | 0x80), (retro_turbo_fire) ? "ON" : "OFF");
          imagename_timer = 50;
          break;

--- a/vice/src/arch/libretro/archdep.c
+++ b/vice/src/arch/libretro/archdep.c
@@ -299,7 +299,7 @@ char *archdep_default_resource_file_name(void)
             if (!string_is_empty(full_path))
             {
                snprintf(content_basename, sizeof(content_basename), "%s", path_basename(full_path));
-               snprintf(content_basename, sizeof(content_basename), "%s", path_remove_extension(content_basename));
+               path_remove_extension(content_basename);
                snprintf(content_vicerc, sizeof(content_vicerc), "%s%s%s.vicerc", SAVEDIR, FSDEV_DIR_SEP_STR, content_basename);
                /* Process "saves/[content].vicerc" */
                if (!ioutil_access(content_vicerc, IOUTIL_ACCESS_R_OK))

